support/elasticsearch2_search.erl

```diff
@@ -349,41 +349,67 @@ source(Query, Default) ->
             [z_convert:to_binary(Field) || Field <- Fields]
     end.
 
-%% @doc Map all sort terms keep asort, sort, zsort order
+%% @doc Map all sort terms keep asort, sort, zsort order. If there is a textual
+%% search then add a sort on _score between 'sort' and 'zsort'.
+%% Do not use the map construct here to ease the serialization into a query argument.
 map_sorts(Query) ->
-    lists:flatten(
-           lists:filtermap(fun(Q) -> map_sort(asort, Q) end, Query)
-        ++ lists:filtermap(fun(Q) -> map_sort(sort, Q) end, Query)
-        ++ lists:filtermap(fun(Q) -> map_sort(zsort, Q) end, Query)
-    ).
+    Sorts = lists:flatten([
+        lists:filtermap(fun(Q) -> map_sort(asort, Q) end, Query),
+        lists:filtermap(fun(Q) -> map_sort(sort, Q) end, Query)
+    ]),
+    lists:flatten([
+        Sorts,
+        case lists:member(<<"_score">>, Sorts) of
+            false -> lists:filtermap(fun(Q) -> map_sort_text(Q) end, Query);
+            true -> []
+        end,
+        lists:filtermap(fun(Q) -> map_sort(zsort, Q) end, Query)]).
+
```

Comment:
Wouldn't it be also ok to write this as:
```
lists:flatten([
        lists:filtermap(fun(Q) -> map_sort(asort, Q) end, Query),
        lists:filtermap(fun(Q) -> map_sort(sort, Q) end, Query),
        case ... end,
        lists:filtermap(fun(Q) -> map_sort(zsort, Q) end, Query)
    ])
```
? Maybe I am missing a nested lists thing.

Reply:
In the case we need the result of the first part, to check if `_score` has been added already.
That is why it has been split up.

Reply:
Oops, that was a sloppy read of me - indeed it looks up score in that case of course.